packages/web/src/components/date/DatePicker.js

```diff
@@ -120,7 +121,9 @@ class DatePicker extends Component {
 		} else if (onChange) {
 			onChange(date || '');
 		} else {
-			this.handleDateChange(date || '');
+			this.setState(state => ({
```

Comment:
It would be great if you can add comments here to state the intent for this key update. Future you will thank you 😸 

```
// this will trigger a remount on the date component
// since it doesn't respect the controlled behavior setting on its own
```